src/main/java/org/springframework/data/redis/serializer/GenericJackson2JsonRedisSerializer.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.redis.serializer;
+
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeInfo.As;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectMapper.DefaultTyping;
+import com.fasterxml.jackson.databind.ser.SerializerFactory;
+
+/**
+ * @author Christoph Strobl
+ * @since 1.6
+ */
+public class GenericJackson2JsonRedisSerializer implements RedisSerializer<Object> {
+
+	private final ObjectMapper mapper;
+
+	/**
+	 * Creates {@link GenericJackson2JsonRedisSerializer} and configures {@link ObjectMapper} for default typing.
+	 */
+	public GenericJackson2JsonRedisSerializer() {
+		this((String) null);
+	}
+
+	/**
+	 * Creates {@link GenericJackson2JsonRedisSerializer} and configures {@link ObjectMapper} for default typing using the
+	 * given {@literal name}. In case of an {@literal empty} or {@literal null} String the default
+	 * {@link JsonTypeInfo.Id#CLASS} will be used.
+	 * 
+	 * @param classPropertyTypeName Name of the JSON property holding type information. Can be {@literal null}.
+	 */
+	public GenericJackson2JsonRedisSerializer(String classPropertyTypeName) {
+
+		this(new ObjectMapper());
+
+		if (StringUtils.hasText(classPropertyTypeName)) {
+			mapper.enableDefaultTypingAsProperty(DefaultTyping.NON_FINAL, classPropertyTypeName);
+		} else {
+			mapper.enableDefaultTyping(DefaultTyping.NON_FINAL, As.PROPERTY);
+		}
+	}
+
+	/**
+	 * Setting a custom-configured {@link ObjectMapper} is one way to take further control of the JSON serialization
+	 * process. For example, an extended {@link SerializerFactory} can be configured that provides custom serializers for
+	 * specific types.
+	 * 
+	 * @param mapper must not be {@literal null}.
+	 */
+	public GenericJackson2JsonRedisSerializer(ObjectMapper mapper) {
```

Comment:
Do we need an additional overload that takes an `ObjectMapper` *and* the type key?